interpreter/text/arrange.ml

```diff
@@ -185,38 +185,47 @@ struct
 end
 
 (* FIXME *)
-module VectorOp =
+module SimdOp =
 struct
-  (* TODO
-  open Ast.FloatOp
-  *)
+  open Ast.SimdOp
 
   let testop xx = fun _ -> failwith "TODO v128"
 
   let relop xx = fun _ -> failwith "TODO v128"
 
-  let unop xx = fun _ -> failwith "TODO v128"
+  let unop xx (op : unop) = match op with
+    | I32x4 Neg -> "i32x4.neg"
+    | _ -> failwith "Unimplemented v128 unop"
 
-  let binop xx = fun _ -> failwith "TODO v128"
+  let binop xx (op : binop) = match op with
+    | I32x4 Add -> "i32x4.add"
+    | I32x4 Sub -> "i32x4.sub"
+    | I32x4 Mul -> "i32x4.mul"
+    | _ -> failwith "Unimplemented v128 unop"
 
   let cvtop xx = fun _ -> failwith "TODO v128"
 end
 
 let oper (intop, floatop, vectop) op =
-  value_type (type_of op) ^ "." ^
-  (match op with
+  (* v128 operations don't need to be prefixed by the type,
+   * each instruction will specify their prefix (shape).
+   *)
+  let prefix = match op with
+  | V128 o -> ""
+  | _ -> value_type (type_of op) ^ "." in
+  let ops = match op with
   | I32 o -> intop "32" o
   | I64 o -> intop "64" o
   | F32 o -> floatop "32" o
   | F64 o -> floatop "64" o
-  | V128 o -> vectop "128" o
-  )
+  | V128 o ->vectop "128" o
```

Comment:
`vectop` should probably be renamed too? Fine for another PR, though

Reply:
Good point, I can make it in this pr, shouldn't have any conflicts...